tests/test_chiral_restraints.py

```diff
@@ -0,0 +1,526 @@
+    samples_per_worker = int(np.ceil(num_samples / num_workers))
+
+    # batches_per_worker = num_samples // num_workers
+    burn_in_batches = 2000
+    frames, _ = simulate(
+        x_min, U_fn, 300.0, np.ones(num_atoms) * 4.0, 500, samples_per_worker + burn_in_batches, num_workers, seed=seed
+    )
+    # (ytz): discard burn in batches
+    frames = frames[:, burn_in_batches:, :, :]
+    # collect over all workers
+    frames = frames.reshape(-1, num_atoms, 3)[:num_samples]
+    # sanity check that we didn't undersample
+    assert len(frames) == num_samples
+    return frames
+
+
+def test_setup_chiral_atom_restraints():
+    mol = Chem.MolFromMolBlock(
+        """
+  Mrv2202 06072215563D
+
+  5  4  0  0  0  0            999 V2000
+    0.0000    0.0000    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
+   -0.3633   -0.5138    0.8900 H   0  0  0  0  0  0  0  0  0  0  0  0
+    1.0900    0.0000    0.0000 H   0  0  0  0  0  0  0  0  0  0  0  0
+   -0.3633    1.0277    0.0000 H   0  0  0  0  0  0  0  0  0  0  0  0
+   -0.3633   -0.5138   -0.8900 H   0  0  0  0  0  0  0  0  0  0  0  0
+  1  2  1  0  0  0  0
+  1  3  1  0  0  0  0
+  1  4  1  0  0  0  0
+  1  5  1  0  0  0  0
+M  END
+$$$$""",
+        removeHs=False,
+    )
+
+    # needs to be batched in order for jax to play nicely
+    x0 = utils.get_romol_conf(mol)
+    normal_restr_idxs = chiral_utils.setup_chiral_atom_restraints(mol, x0, 0)
+
+    x0_inverted = [x0[0], x0[2], x0[1], x0[3], x0[4]]
+
+    inverted_restr_idxs = chiral_utils.setup_chiral_atom_restraints(mol, x0_inverted, 0)
+
+    # check the sign of the resulting idxs
+    k = 1000.0
+    assert np.all(np.asarray(U_chiral_atom_batch(x0, normal_restr_idxs, k)) == 0)
+    assert np.all(np.asarray(U_chiral_atom_batch(x0, inverted_restr_idxs, k)) > 0)
+    assert np.all(np.asarray(U_chiral_atom_batch(x0_inverted, normal_restr_idxs, k)) > 0)
+    assert np.all(np.asarray(U_chiral_atom_batch(x0_inverted, inverted_restr_idxs, k)) == 0)
+
+
+def test_setup_chiral_bond_restraints():
+    mol = Chem.MolFromMolBlock(
+        """
+  Mrv2202 06082215002D
+
+  5  4  0  0  0  0            999 V2000
+   -0.5335   -0.0247    0.0000 F   0  0  0  0  0  0  0  0  0  0  0  0
+    0.0285   -0.4092    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
+   -0.0697   -1.2953    0.0000 Cl  0  0  0  0  0  0  0  0  0  0  0  0
+    0.6376   -0.0935    0.0000 N   0  0  0  0  0  0  0  0  0  0  0  0
+    1.2163   -0.3988    0.0000 F   0  0  0  0  0  0  0  0  0  0  0  0
+  2  4  2  0  0  0  0
+  4  5  1  0  0  0  0
+  1  2  1  0  0  0  0
+  2  3  1  0  0  0  0
+M  END
+$$$$
+""",
+        removeHs=False,
+    )
+
+    # needs to be batched in order for jax to play nicely
+    x0 = utils.get_romol_conf(mol)
+    src_atom = 1
+    dst_atom = 3
+    normal_restr_idxs, signs = chiral_utils.setup_chiral_bond_restraints(mol, x0, src_atom, dst_atom)
+
+    x0_inverted = [x0[0], x0[1], x0[2], x0[3], [0.07922, 0.06280, 0.0000]]
```

Comment:
Nit: this flips the conformer from
<img width="173" alt="image" src="https://user-images.githubusercontent.com/5759036/173100794-940e3f5f-6e58-4723-95e7-6f0a928855c1.png">
 to 
<img width="173" alt="image" src="https://user-images.githubusercontent.com/5759036/173100823-41073902-6dd3-4de7-aa2b-b0a6a138e86e.png">
which does change the dihedral (0,1,3,4) dihedral angle from -pi to 0 (but also changes an angle)

Consider using

```python
from rdkit.Chem import rdMolTransforms

assert np.abs(rdMolTransforms.GetDihedralRad(mol.GetConformer(0), 0, 1, 3, 4)) == np.pi
rdMolTransforms.SetDihedralRad(mol.GetConformer(0), 0, 1, 3, 4, 0.0)
```
or similar instead?

Reply:
(Alternately, could have RDKit generate conformers for 'Cl\C(F)=N/F'` vs. `'Cl/C(F)=N/F'`)